### DISCUSSION TITLE
ASUS ROG Ally

### DISCUSSION THREAD

**Discussion by gloa7** [seerge/g-helper#461] [Q&A]
Curious if there's a possibility of compatibility with the new ASUS ROG Ally with G-Helper?

**Comment by seerge** [comment]
@gloatoriginal i don't know, as i don't have a device :) 

Mode selection / fans / ppts , also screen refresh rate changer (as it just uses windows API) and backlight colors should work (probably). I can imagine XG Mobile control can also work/

dGPU disabling definetely won't work, as there is no dGPU there ;)

Moving this to discussions as this is not an "issue" with the app.

**Reply by Valkirie** [reply]
Hey team, I'm working on a application to provide global windows support to  handheld gaming devices, such a Rog Ally. I've made progress lately and was "almost" able to figure out how to read and intercept all custom buttons. Except the one from the back of the device.

Right now I'm listening to inputs sent by the ASUS custom HID device. But both back buttons are reporting the same value. Unless you start Armory Crate SE which then seams to send a signal to the device, preventing the two back buttons to send signals to the HID device and instead send appropriate XInput buttons based on AC SE controller mapping configuration.

I don't know where to search and what to search exactly now.

Code involved: https://github.com/Valkirie/HandheldCompanion/blob/def24195fc20f25ed22742ae0a686459828d8afe/ControllerCommon/Devices/ASUS/ROGAlly.cs#L115

Happy to get thoughts and ideas here.

**Comment by Inaiso** [comment]
So has anyone tried ghelper on ally? How does it function? Thinking about getting it, but i dislike AC too much.

**Reply by seerge** [reply]
I have rog ally :) all main things work there (like modes / power limits / fan curves / refresh rates and even backlight control ;)

But keep in mind that ROG Ally is a handheld, and therefore has lots of things that laptops don't have (like built in gamepad / etc) 

And AC there also offers all bells and whistles like binding for gamepad buttons, deadzone control, game launcher, etc, that simply don't exist on laptops and therefore in g-helper :) But it still feels buggy as for me

**Reply by Valkirie** [reply]
Here is how I'm listening to Ally custom buttons. It's not ideal as both back buttons seams to report the same value... Maybe you'll find a workaround. 

https://github.com/Valkirie/HandheldCompanion/blob/8adbfa52a04e4d30b47d9300cd30d8718925f1d9/ControllerCommon/Devices/ASUS/ROGAlly.cs#L76

**Reply by seerge** [reply]
@Valkirie i do almost same from the beginning :) (g-helper even logs key ids). 
It just don't process some of new keys (but armoury-key works - it toggles main app) :) 

Does your approach have issues when device goes to sleep / wakes up ?

**Reply by Valkirie** [reply]
I think I heard someone complaining about it. I had to send back my loan sample to asus so won't be able to test before July. Can you let me know if you make discoveries related to the Ally? :)
